components/apps_form/apps_form_field/apps_form_field.tsx

```diff
@@ -45,49 +40,15 @@ export interface Props {
 }
 
 export default class AppsFormField extends React.PureComponent<Props> {
-    private providers: Provider[] = [];
-
     static defaultProps = {
         listComponent: ModalSuggestionList,
     };
 
-    constructor(props: Props) {
-        super(props);
-        this.setProviders();
-    }
-
     handleSelected = (selected: AppSelectOption | UserProfile | Channel) => {
-        const {name, field, onChange} = this.props;
-
-        if (field.type === AppFieldTypes.USER) {
-            const user = selected as UserProfile;
-            let selectedLabel = user.username;
-            if (this.props.teammateNameDisplay) {
-                selectedLabel = displayUsername(user, this.props.teammateNameDisplay);
-            }
-            const option = {label: selectedLabel, value: user.id};
-            onChange(name, option);
```

Comment:
@prashant-15 I've tested out this PR, and it works great! 

I've found one issue that makes this new functionality inconsistent with existing behavior. Notice this line I'm commenting on here. We used to convert the user and channel objects into a `{label: string, value: string}` object, but we are now submitting them as full user and channel objects.

Are you able to look into this and change the functionality to reflect the previous behavior? The `as AppSelectOption` is sort of the culprit here, since we aren't ensuring the type is correct before this statement.

Thanks for all of the effort on this @prashant-15! I think we're close to getting this out the door :+1:

Reply:
this should still work as earlier right since we already adapt this before passing to `AppFormSelectField` (https://github.com/mattermost/mattermost-webapp/blob/533d8d2580d8eb5ab1c2df6a3ab9e95a9e5f0377/components/apps_form/apps_form_field/apps_form_select_field.tsx#L98) so in OnChange the behaviour will be as it was?

Reply:
@prashant-15 We specifically want the object that gets submitted to the backend to only contain `label` and `value`, and this implementation is currently providing more fields than these two fields